datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireViewModel.kt

```diff
@@ -418,9 +439,40 @@ internal class QuestionnaireViewModel(application: Application, state: SavedStat
           )
           .map { it.copy() }
       unpackRepeatedGroups(this@QuestionnaireViewModel.questionnaire)
+      addTimeStampValue(item, "submission-timestamp")
     }
   }
 
+    private fun addTimeStampValue(items: List<QuestionnaireResponseItemComponent>, itemLinkId : String) {
+        if (shouldAddTimestampItems) {
+            items.forEach { item ->
+                if (item.linkId == itemLinkId) {
+                    item.answer = arrayListOf<QuestionnaireResponseItemAnswerComponent?>().apply {
+                        add(QuestionnaireResponseItemAnswerComponent().apply {
+                            value = DateTimeType(Date().toTimeZoneString())
+                        })
+                    }
+                }
+            }
+        }
+    }
```

Comment:
```suggestion
    private fun addTimeStampValue(items: List<QuestionnaireResponseItemComponent>, itemLinkId : String) {
            items.forEach { item ->
                if (item.linkId == itemLinkId) {
                    item.answer = arrayListOf<QuestionnaireResponseItemAnswerComponent?>().apply {
                        add(QuestionnaireResponseItemAnswerComponent().apply {
                            value = DateTimeType(Date().toTimeZoneString())
                        })
                    }
                }
            }
        }
```

Or maybe rename it to describe the intent better.